### PR TITLE
Include ad title in bulletin board search

### PR DESCRIPTION
**Pull request by zonkmachine**
Include ad title in search on the bulletin board.

Motivation: I just found it more practical/correct after interpreting an earlier search result as a wee bit odd.

### CODE REVIEW

File: data/pigui/modules/station-view/02-bulletinBoard.lua [hunk 1]
```diff
@@ -56,6 +56,10 @@ local function refresh()
 
 	for ref,ad in pairs(ads or {}) do
 		if searchText == ""
+			or searchText ~= "" and string.find(
```
Comment (zonkmachine) [inline data/pigui/modules/station-view/02-bulletinBoard.lua:59]: Isn't `searchText ~= "" and` in this and the next paragraph redundant after the test on line 58?

Comment (zonkmachine) [reply data/pigui/modules/station-view/02-bulletinBoard.lua:59]: I mean you could simplify it as such:

```lua
        for ref,ad in pairs(ads or {}) do
                if searchText == ""
-                       or searchText ~= "" and string.find(
+                       or string.find(
                                string.lower(ad.title),
                                string.lower(searchText),
                                1, true)
-                       or searchText ~= "" and string.find(
+                       or string.find(
                                string.lower(ad.description),
                                string.lower(searchText),
                                1, true)

```

Comment (Web-eWorks) [reply data/pigui/modules/station-view/02-bulletinBoard.lua:59]: @zonkmachine is correct, `searchText ~= ""` is implied by the first condition of the if statement evaluating to false. The only way this breaks is if `nil` is passed as `searchText`, which can be handled by changing the first condition to `if not searchText or searchText == ""`.

**Review by Web-eWorks — APPROVED** [review]
Looks good, if the code has been tested then this can be squashed and merged.

**Comment by impaktor** [comment]
If someone has tested this, please holler

**Pull request merged**